presto-native-execution/src/test/java/com/facebook/presto/nativeworker/PrestoNativeQueryRunnerUtils.java

```diff
@@ -92,101 +91,249 @@ public class PrestoNativeQueryRunnerUtils
+        public HiveQueryRunnerBuilder setEnableRuntimeMetricsCollection(boolean enableRuntimeMetricsCollection)
+        {
+            this.enableRuntimeMetricsCollection = enableRuntimeMetricsCollection;
+            return this;
+        }
 
-    public static QueryRunner createJavaQueryRunner(String storageFormat)
-            throws Exception
-    {
-        return createJavaQueryRunner(Optional.of(getNativeQueryRunnerParameters().dataDirectory), storageFormat, true);
+        public HiveQueryRunnerBuilder setCoordinatorSidecarEnabled(boolean coordinatorSidecarEnabled)
+        {
+            this.coordinatorSidecarEnabled = coordinatorSidecarEnabled;
+            if (coordinatorSidecarEnabled) {
+                this.extraProperties.putAll(getNativeSidecarProperties());
+            }
+            return this;
+        }
+
+        public HiveQueryRunnerBuilder setStorageFormat(String storageFormat)
+        {
+            this.storageFormat = storageFormat;
+            this.hiveProperties.put("hive.storage-format", storageFormat);
+            return this;
+        }
+
+        public HiveQueryRunnerBuilder setAddStorageFormatToPath(boolean addStorageFormatToPath)
+        {
+            this.addStorageFormatToPath = addStorageFormatToPath;
+            return this;
+        }
+
+        public HiveQueryRunnerBuilder setCacheMaxSize(Integer cacheMaxSize)
+        {
+            this.cacheMaxSize = cacheMaxSize;
+            return this;
+        }
+
+        public HiveQueryRunnerBuilder setExtraProperties(Map<String, String> extraProperties)
+        {
+            this.extraProperties.putAll(extraProperties);
+            return this;
+        }
+
+        public QueryRunner build()
+                throws Exception
+        {
+            Optional<BiFunction<Integer, URI, Process>> externalWorkerLauncher = Optional.empty();
+            if (this.useExternalWorkerLauncher) {
+                externalWorkerLauncher = getExternalWorkerLauncher("hive", serverBinary, cacheMaxSize, remoteFunctionServerUds,
+                        failOnNestedLoopJoin, coordinatorSidecarEnabled, enableRuntimeMetricsCollection, enableSsdCache);
+            }
+            return HiveQueryRunner.createQueryRunner(
+                    ImmutableList.of(),
+                    ImmutableList.of(),
+                    extraProperties,
+                    extraCoordinatorProperties,
+                    security,
+                    hiveProperties,
+                    Optional.ofNullable(workerCount),
+                    Optional.of(Paths.get(addStorageFormatToPath ? dataDirectory.toString() + "/" + storageFormat : dataDirectory.toString())),
+                    externalWorkerLauncher,
+                    tpcdsProperties);
+        }
     }
 
-    public static QueryRunner createJavaQueryRunner(String storageFormat, boolean addStorageFormatToPath)
-            throws Exception
+    public static IcebergQueryRunnerBuilder nativeIcebergQueryRunnerBuilder()
```

Comment:
Similar to the comment for HiveQueryRunnerBuilder... we have a constructor for IcebergQueryRunnerBuilder which takes an enum for Java or Native builder.